### DISCUSSION TITLE
Error rendering template. Review log for more info.

### DISCUSSION THREAD

**Discussion by scarolan** [tsightler/ring-mqtt#557] [Support Q&A] [answered]
Following the installation and setup steps, I got my Ring Doorbell to be discovered by ring-mqtt.  Under the Info section I find:

```
Stream Source
rtsp://deadbeef-ring-mqtt:8554/94XXXXXXX8d6_live

Still Image URL
https://homeassistant:8123{{ states.camera.front_door_snapshot.attributes.entity_picture }}
```

I tried updating the homeassistant URL to plain http://homeassistant.lan, which is what works for everything else.

I only get this error rendering template message when I try to connect.

I ran an nmap scan against my host and there's nothing listening on port 8554 either.  Any ideas how to troubleshoot this?

**Comment by scarolan** [comment, marked as answer]
Please see @tsightler 's response right here:

https://github.com/tsightler/ring-mqtt/discussions/557#discussioncomment-4951156

TL;DR - the network name in the rtsp URL is local to the docker host running Home Assistant, therefore it is normal for it not to be resolvable from your laptop or workstation.

**Reply by scarolan** [reply]
Also tried hardcoding the IP address of the home assistant machine, no luck there either.

rtsp://192.168.86.40:8554/91d3612348d6_live

**Reply by tsightler** [reply]
You're a little all over the place here, and your obfuscating things that I would need to see to be the most useful, but the documentation is pretty clear that the still image URL is just the best guess it could make, making various assumptions about the HA environment as it doesn't have any way to know the specific details, especially for things like renamed entities.

> rtsp://deadbeef-ring-mqtt:8554/94XXXXXXX8d6_live

I'm going to hope that this is actually rtsp://03cabcc9-ring-mqtt:8554/94XXXXXXX8d6_live and you've just obfuscated the ID portion of the hostname.  The 03cabcc9-ring-mqtt is the name of the docker container as assigned by Home Assistant using an algorithm based on the name of the repository, there's nothing secret about it.

> How exactly is the hostname supposed to get resolved?

Home Assistant is basically just a bunch of docker containers running on an OS, and as a basic feature of Docker, it provides resolution of the container names from the Docker host as well as between containers on the same host/Docker network.  Of course those hostnames are not resolvable outside of that host, but this isn't needed since both the HA container and ring-mqtt container are on the same host.  As noted above, Home Assistant uses a predictable algorithm for the hostnames of the containers so that it doesn't matter what IP address Docker assigns to the hosts, you can always reference a given addon by its name.

If you want to use the IP of your host for RTSP you'd need to configure the port the be exposed (the RTSP port is available only within the Docker network by default, because that's all that's needed to view the stream int he Docker Ui).  Exposing the port on the host is only required if you want to connect to the stream using some external client, like VLC.

**Reply by scarolan** [reply]
How do I mark your reply as the correct answer?  I'm not seeing the button on the nested reply...

**Reply by tsightler** [reply]
Honestly, no idea, don't see any obvious way to make a reply an answer.  Weird, IMO.  Maybe answers are supposed to be new comments and not replies.

**Reply by scarolan** [reply]
Ok, I'll make a new top level reply and point users to your response.

**Comment by scarolan** [comment]
Thank you for the detailed reply, @tsightler.  My setup is a bit weird, I'm running Home Assistant on a Hyper-V VM (long story, there's a whole thread in the HA forums).  It's on a bridged network which I suspect may be causing some problems.  

I've ordered a dedicated device that can sit directly on the network which I'm hoping will resolve some of these network and DNS issues.  

One other question - is there any harm running your add-on alongside the Ring add-on from the main HA repository?  I have them both installed side by side, but that plugin doesn't expose any of the sensors or individual devices.

**Reply by tsightler** [reply]
I wouldn't think running in a VM would matter much, my development system is a VMware VM on a bridged network.  Assuming you are running Home Assistant OS and this as an addon, the Docker network exist internally to the host regardless of the external network.  I doubt your environment would work properly at all if the Docker internal name resolution wasn't working.

There's no issue running the two together, other than some potential confusion since the same devices might appear twice in some cases.

**Reply by scarolan** [reply]
Thanks for the info, and great work on your plugin.  I'm still in a 'dev/test' mode with home assistant, waiting on some hardware to build the production version.
